scripts/zkai_tablero_hybrid_zkml_boundary_gate.py

```
@@ -0,0 +1,799 @@
+
+def require_str(value: Any, label: str) -> str:
+    if not isinstance(value, str) or not value:
+        raise TableroHybridBoundaryError(f"{label} must be a non-empty string")
+    return value
+
+
+def row_by_component(rows: list[Any], component: str) -> dict[str, Any]:
+    for row in rows:
+        if isinstance(row, dict) and row.get("component") == component:
+            return row
+    raise TableroHybridBoundaryError(f"missing component row: {component}")
```

Ruff (0.15.12)

[warning] Avoid specifying long messages outside the exception class

(TRY003)